Monika After Story/game/script-compliments.rpy

```diff
@@ -888,6 +889,174 @@ label mas_compliment_humor_3:
     m 1hubsu "[humor_quip]"
     return
 
+init 5 python:
+    addEvent(
+        Event(
+            persistent._mas_compliments_database,
+            eventlabel="mas_compliment_missed",
+            prompt="I missed you!",
+            unlocked=False,
+            conditional="not mas_globals.returned_home_this_sesh",
+            action=EV_ACT_UNLOCK
```

Comment:
>The new system will take care of the rest.

By new system, do you mean this suggested change right here, or is there some other file you changed? With all of the suggested changes (including removing line 904/905), it still unlocks/unhides regardless of the conditionals.

Reply:
By that I mean fd4fd55. If you commit the req changes, it shouldn't unlock/lock at all. Instead when the `conditional` evaluates to `False`, this compliment won't appear in the menu.

Reply:
Thanks, I actually didn't realize you could commit to my fork! I'll test with your changes.

Reply:
Only to this one branch since you opened a PR from it. I did that because it was easier than explaining in review what I wanted. I also did test it, so should be fine (although, obviously I could miss something).

Reply:
Now I'm thinking... We could check sesh length in the `conditional`, so we wouldn't need to hide this compliment from somewhere else.

Reply:
Did you intend to commit only comments to the definitions.rpy file?

Reply:
>Now I'm thinking... We could check sesh length in the conditional, so we wouldn't need to hide this compliment from somewhere else.

Regarding this idea, I thought that the init block runs only once, and if I'm right then session length would always be ~0?

Reply:
We won't do it in init blocks, we'd do it in `conditional`. With my changes we check them every time you open the menu.

Reply:
Resolved this in https://github.com/Monika-After-Story/MonikaModDev/pull/7419/commits/10aded4ee54487a9a6608935f06cd19427da81f4.